01-data-structures/02-stacks-and-queues/mystack/mystack.rb

```diff
@@ -4,14 +4,34 @@ class MyStack
   def initialize
     @stack = Array.new
     self.top = nil
+    @pointer = -1
   end
 
   def push(item)
+    @pointer += 1
+    @stack[@pointer] = item
+    self.top = item
   end
 
   def pop
+    temp = @stack[@pointer]
+
+    @pointer -= 1
+
+    if @pointer < 0
+      self.top = nil
+    else
+      self.top = @stack[@pointer]
+    end
+
+    return temp
   end
 
   def empty?
+    if self.top == nil
+      return true
+    else
+      return false
+    end
```

Comment:
This kind of logic always amuses me; it is one of the most common things I see. Your code is basically saying "if `self.top == nil` evaluates to `true`, return the constant `true`. If it evaluates to `false`, return the constant `false`".